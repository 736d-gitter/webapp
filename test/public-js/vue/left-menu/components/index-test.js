'use strict';

const { createLocalVue, shallowMount } = require('@vue/test-utils');
const Vuex = require('vuex');

const createStore = require('../../../../../public/js/vue/store').default;
const actions = require('../../../../../public/js/vue/store/actions');
const Index = require('../../../../../public/js/vue/left-menu/components/index.vue');

let wrapper;
let stubbedActions = {};
function factory(propsData = {}, extendStore = () => {}) {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  Object.keys(actions).forEach(actionKey => {
    stubbedActions[actionKey] = jest.fn();
  });

  const store = createStore({
    actions: stubbedActions
  });
  extendStore(store);

  wrapper = shallowMount(Index.default, {
    localVue,
    store,
    propsData
  });
}

describe('left-menu index', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  describe('all state', () => {
    it('matches snapshot', () => {
      factory({}, store => {
        store.state.leftMenuState = 'all';
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('matches snapshot with some rooms', () => {
      factory({}, store => {
        store.state.leftMenuState = 'all';

        store.state.roomMap = {
          1: { id: 1, lastAccessTime: 1 },
          2: { id: 2, lastAccessTime: 1 },
          3: { id: 3, lastAccessTime: 1 }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('people state', () => {
    it('matches snapshot', () => {
      factory({}, store => {
        store.state.leftMenuState = 'people';
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('search state', () => {
    it('matches snapshot', () => {
      factory({}, store => {
        store.state.leftMenuState = 'search';
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('pinning and expanding', () => {
    it('not pinned and collapse', () => {
      factory({}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = false;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not pinned and expanded', () => {
      factory({}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = true;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('calls store action "toggleLeftMenu" after mouse leaves', () => {
      factory({}, store => {
        store.state.leftMenuPinnedState = false;
        store.state.leftMenuExpandedState = true;
      });
      wrapper.find({ ref: 'root' }).trigger('mouseleave');

      expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(
        expect.anything(),
        false,
        undefined
      );
    });
  });
});
