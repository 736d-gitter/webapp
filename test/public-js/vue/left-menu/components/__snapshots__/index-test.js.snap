// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`left-menu index all state matches snapshot 1`] = `
<div
  class="root js-left-menu-root"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <ul
        class="room-list"
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index all state matches snapshot with some rooms 1`] = `
<div
  class="root js-left-menu-root"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <ul
        class="room-list"
      >
        <list-item-stub
          item="[object Object]"
          type="all"
        />
        <list-item-stub
          item="[object Object]"
          type="all"
        />
        <list-item-stub
          item="[object Object]"
          type="all"
        />
      </ul>
    </section>
  </section>
</div>
`;

exports[`left-menu index people state matches snapshot 1`] = `
<div
  class="root js-left-menu-root"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <!---->
       
      <h2
        class="room-list-title"
      >
        People
      </h2>
       
      <ul
        class="room-list"
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index pinning and expanding not pinned and collapse 1`] = `
<div
  class="root js-left-menu-root unpinned"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <ul
        class="room-list"
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index pinning and expanding not pinned and expanded 1`] = `
<div
  class="root js-left-menu-root unpinned expanded"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <h2
        class="room-list-title"
      >
        All conversations
      </h2>
       
      <!---->
       
      <ul
        class="room-list"
      />
    </section>
  </section>
</div>
`;

exports[`left-menu index search state matches snapshot 1`] = `
<div
  class="root js-left-menu-root"
>
  <header
    class="header"
  >
    <section
      class="header-minibar layout-minibar"
    >
      <svg
        class="logo-gitter-sign"
        version="1.1"
        viewBox="0 0 18 25"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <rect
          height="10"
          width="2"
          x="15"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="10"
          y="5"
        />
         
        <rect
          height="20"
          width="2"
          x="5"
          y="5"
        />
         
        <rect
          height="15"
          width="2"
        />
      </svg>
    </section>
     
    <section
      class="header-main-menu layout-main-menu"
    >
      <span
        class="logo-text"
      />
    </section>
  </header>
   
  <section
    class="body"
  >
    <section
      class="layout-minibar"
    >
      <menu-bar-body-stub />
    </section>
     
    <section
      class="body-main-menu layout-main-menu"
    >
      <div
        class="hide"
      >
        
        initialTestValue
      
      </div>
       
      <search-body-stub />
    </section>
  </section>
</div>
`;
